src/generic/Matrix.jl

```diff
@@ -28,32 +28,40 @@ export MatrixSpace, fflu!, fflu, solve_triu, isrref, charpoly_danilevsky!,
 #
 ###############################################################################
 
-function similar(x::Mat{T}) where T <: RingElement
-   R = base_ring(x)
-   M = similar(x.entries)
+@doc Markdown.doc"""
+    similar(x::Mat, [R::Ring=base_ring(x)]
+    similar(x::Mat, [R::Ring=base_ring(x)], r::Int, c::Int)
```

Comment:
Also, why do you put square brackets around the second parameter? It's not in the actual function prototype.

Reply:
It's how it's done in the `similar` docstring in `Base`. It's meant to represent optional positional argument, followed by non-optional arguments. For the first signature, I agree it can be removed. For the 2nd one, do you prefer that I duplicate it into two standard signatures?